packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,585 @@
+      : boundingBox = _data['left'] != null
+            ? Rectangle<int>(
+                _data['left'],
+                _data['top'],
+                _data['width'],
+                _data['height'],
+              )
+            : null,
+        rawValue = _data['raw_value'],
+        displayValue = _data['display_value'],
+        format = BarcodeFormat._(_data['format']),
+        cornerPoints = _data['points'] == null
+            ? null
+            : _data['points']
+                .map<Point<int>>((dynamic item) => Point<int>(
+                      item[0],
+                      item[1],
+                    ))
+                .toList(),
+        valueType = BarcodeValueType.values.elementAt(_data['value_type']),
+        email = _data['email'] == null ? null : BarcodeEmail._(_data['email']),
+        phone = _data['phone'] == null ? null : BarcodePhone._(_data['phone']),
+        sms = _data['sms'] == null ? null : BarcodeSMS._(_data['sms']),
+        url = _data['url'] == null ? null : BarcodeURLBookmark._(_data['url']),
+        wifi = _data['wifi'] == null ? null : BarcodeWiFi._(_data['wifi']),
+        geoPoint = _data['geo_point'] == null
+            ? null
+            : BarcodeGeoPoint._(_data['geo_point']),
+        contactInfo = _data['contact_info'] == null
+            ? null
+            : BarcodeContactInfo._(_data['contact_info']),
+        calendarEvent = _data['calendar_event'] == null
+            ? null
+            : BarcodeCalendarEvent._(_data['calendar_event']),
+        driverLicense = _data['driver_license'] == null
+            ? null
+            : BarcodeDriverLicense._(_data['driver_license']);
+
+  /// Gets the bounding rectangle of the detected barcode.
+  ///
+  /// Returns null if the bounding rectangle can not be determined.
+  final Rectangle<int> boundingBox;
+
+  /// Returns barcode value as it was encoded in the barcode.
+  ///
+  /// Structured values are not parsed, for example: 'MEBKM:TITLE:Google;URL://www.google.com;;'.
+  ///
+  /// Returns null if nothing found.
+  final String rawValue;
+
+  /// Returns barcode value in a user-friendly format.
+  ///
+  /// May omit some of the information encoded in the barcode.
+  /// For example, if getRawValue() returns 'MEBKM:TITLE:Google;URL://www.google.com;;',
+  /// the display_value might be '//www.google.com'.
+  /// If valueFormat==TEXT, this field will be equal to getRawValue().
+  ///
+  /// This value may be multiline, for example, when line breaks are encoded into the original TEXT barcode value.
+  /// May include the supplement value.
+  ///
+  /// Returns null if nothing found.
+  final String displayValue;
+
+  /// The barcode format, for example [BarcodeFormat.EAN13].
+  final BarcodeFormat format;
+
+  /// The four corner points in clockwise direction starting with top-left.
+  ///
+  /// Due to the possible perspective distortions, this is not necessarily a rectangle.
+  final List<Point<int>> cornerPoints;
```

Comment:
For the classes that use a `List` variable, we don't want the user to be able to modify the list. `final` only keeps `cornerPoints` from being assigned a new `List`.

I typically just create a new list with a getter:
```dart
final List<Point<int>> _cornerPoints;

List<Point<int>> get cornerPoints => List<Point<int>>.from(_cornerPoints);
```